File: lib/gemini.py
import os
from typing import List
import google.generativeai as genai

# Configurar la API key
api_key = os.environ.get('GOOGLE_API_KEY')
if not api_key:
    raise ValueError('GOOGLE_API_KEY no está configurada en las variables de entorno')

genai.configure(api_key=api_key)

def get_embeddings(text: str) -> List[float]:
    """
    Generar embeddings usando el modelo embedding-001 de Gemini
    """
    try:
        # Usar la función embed_content directamente
        result = genai.embed_content(
            model='models/embedding-001',
            content=text
        )
        embedding = result['embedding']
        
        if not embedding or len(embedding) != 768:
            raise ValueError('El embedding generado no tiene la dimensión correcta (768)')
        
        return embedding
        
    except Exception as error:
        print(f'Error al obtener embeddings de Gemini: {error}')
        raise error 

File: lib/gemini.ts
import { GoogleGenerativeAI } from '@google/generative-ai';

if (!process.env.GOOGLE_API_KEY) {
  throw new Error('GOOGLE_API_KEY no está configurada en las variables de entorno');
}

const genAI = new GoogleGenerativeAI(process.env.GOOGLE_API_KEY);

export async function getEmbeddings(text: string): Promise<number[]> {
  try {
    const model = genAI.getGenerativeModel({ model: 'embedding-001' });
    const result = await model.embedContent(text);
    const embedding = result.embedding.values;
    
    if (!embedding || embedding.length !== 768) {
      throw new Error('El embedding generado no tiene la dimensión correcta');
    }
    
    return embedding;
  } catch (error) {
    console.error('Error al obtener embeddings de Gemini:', error);
    throw error;
  }
} 